Src/main.c

```diff
@@ -67,110 +67,148 @@ void SystemClock_Config(void);
-{
+int main(void) {
+
+    /* USER CODE BEGIN 1 */
+
+    /* USER CODE END 1 */
+
+    /* MCU Configuration----------------------------------------------------------*/
+
+    /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
+    HAL_Init();
+
+    /* USER CODE BEGIN Init */
+
+    /* USER CODE END Init */
+
+    /* Configure the system clock */
+    SystemClock_Config();
+
+    /* USER CODE BEGIN SysInit */
 
-  /* USER CODE BEGIN 1 */
+    /* USER CODE END SysInit */
 
-  /* USER CODE END 1 */
+    /* Initialize all configured peripherals */
+    MX_GPIO_Init();
+    MX_ADC_Init();
+    MX_TIM16_Init();
+    MX_USART1_UART_Init();
 
-  /* MCU Configuration----------------------------------------------------------*/
+    /* USER CODE BEGIN 2 */
+    debug("\n\nBootcamp starting up...");
+    debug("Compiled on %s at %s", __DATE__, __TIME__);
 
-  /* Reset of all peripherals, Initializes the Flash interface and the Systick. */
-  HAL_Init();
 
-  /* USER CODE BEGIN Init */
+    HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
 
-  /* USER CODE END Init */
+    HAL_ADC_MspInit(&hadc)
 
-  /* Configure the system clock */
-  SystemClock_Config();
 
-  /* USER CODE BEGIN SysInit */
+    HAL_TIM_Base_MspPostInit(&htim16) // I think this is the right one to go first
+    // MSP POST should init the pot checker?
+    HAL_TIM_Base_MspInit(&htim16) // starts up clock timer
+    /* USER CODE END 2 */
 
-  /* USER CODE END SysInit */
+    /* Infinite loop */
+    /* USER CODE BEGIN WHILE */
 
-  /* Initialize all configured peripherals */
-  MX_GPIO_Init();
-  MX_ADC_Init();
-  MX_TIM16_Init();
-  MX_USART1_UART_Init();
+    /* Note:
+     * input clock freq 48Mhz <- I have no clue what this means
+     * Pot voltage 0-3.3V
+     * pwm mod 1-2ms ot @50hz*/
 
-  /* USER CODE BEGIN 2 */
-  debug("\n\nBootcamp starting up...");
-  debug("Compiled on %s at %s", __DATE__, __TIME__);
+    unsigned int lowest = 1;
+    unsigned int highest = 2;
+    int cycleConst = 3000;
 
+    while (1) {
+        double adc_val = HAL_ADC_GetValue(&hadc); // if I read the docs right this should be it
+        // this should max at 1024-1?
+        // needs to be turned into on_cycles for 2ms = 60 000 ticks?
+        // pwm @ 50hz so
+        // 2ms -> idek we'll just come up with numbers
 
-  HAL_GPIO_WritePin(LED1_GPIO_Port, LED1_Pin, GPIO_PIN_SET);
-  /* USER CODE END 2 */
+        float oncycles = adc_val / 1023; // basic conversion to decimal
+        //I don't actually know if this init works in c but we'll go with it
 
-  /* Infinite loop */
-  /* USER CODE BEGIN WHILE */
-  while (1)
-  {
-  /* USER CODE END WHILE */
+        oncycles *= (highest-lowest); // this doesn't change here but in the future it could.
+        // (IDK I COPIED THIS FROM THE INTERNET)
 
-  /* USER CODE BEGIN 3 */
+        oncycles *= cycleConst;
 
-  }
-  /* USER CODE END 3 */
+        oncycles += lowest * cycleConst; // add 1ms bc that's the minimummmmm
```

Comment:
Math looks good to me. Would it be possible to do it on one line though? 

Also, another thing to consider once you make oncycles an integer: truncation when dividing integers (https://stackoverflow.com/questions/3602827/what-is-the-behavior-of-integer-division)